### PR TITLE
Add an OpenTelemetry gRPC access logger (just gRPC handling)

### PR DESCRIPTION
**Pull request by itamarkam**
Commit Message: Add an [OpenTelemetry](http://opentelemetry.io) access logger (only gRPC handling)
Additional Description: This PR adds a dependency on the OpenTelemetry protos used for logging. The actual part where the log entry is filled ("Common::ImplBase::emitLog") isn't implemented yet, and there's no API to activate this logger.
Risk Level: Low (this is not linked to the Envoy binary as nothing creates this logger)
Testing: Unit tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by yanavlasov** [comment]
@itamarkam thanks this looks good. I would suggest moving the `grpc_ot_access_log_impl.*` and corresponding tests into its own directory (i.e. `source/extensions/access_loggers/open_telemetry` or some such), since this is a distinct logger from Envoy's gRPC logger.
Please also register the new logger in the `source/extensions/extensions_build_config.bzl`

**Comment by itamarkam** [comment]
> @itamarkam thanks this looks good. I would suggest moving the `grpc_ot_access_log_impl.*` and corresponding tests into its own directory (i.e. `source/extensions/access_loggers/open_telemetry` or some such), since this is a distinct logger from Envoy's gRPC logger.

Done. I also moved it from the GrpcCommon namespace to its own OpenTelemetry namespace, and removed the "OpenTelemetry" prefix from all the access logger and cache classes.

> Please also register the new logger in the `source/extensions/extensions_build_config.bzl`
This is not yet a configurable extension - it's missing the proto definition on how to configure it and the logger that actually fills a log entry based on the request/response info. I was planning on adding those 2 in a separate PR to reduce the review complexity. WDYT?

**Comment by itamarkam** [comment]
> CC @envoyproxy/dependency-shepherds: Your approval is needed for changes made to `(bazel/.*repos.*\.bzl)|(bazel/dependency_imports\.bzl)|(api/bazel/.*\.bzl)|(.*/requirements\.txt)|(.*\.patch)`.

/assign htuch

File: api/bazel/repository_locations.bzl [hunk 1]
```diff
@@ -96,4 +96,15 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
+    opentelemetry_proto = dict(
+        project_name = "OpenTelemetry Proto",
+        project_desc = "Language Independent Interface Types For OpenTelemetry",
+        project_url = "https://github.com/open-telemetry/opentelemetry-proto",
+        version = "0.6.0",
+        sha256 = "08f090570e0a112bfae276ba37e9c45bf724b64d902a7a001db33123b840ebd6",
+        strip_prefix = "opentelemetry-proto-{version}",
+        urls = ["https://github.com/open-telemetry/opentelemetry-proto/archive/v{version}.tar.gz"],
+        release_date = "2020-10-27",
+        use_category = ["api"],
```
Comment (moderation) [inline api/bazel/repository_locations.bzl:108]: `0.7.0` was released a week ago - can we move to that latest version for this PR? https://github.com/open-telemetry/opentelemetry-proto/releases/tag/v0.7.0

**Review by moderation — COMMENTED** [review]
Question re: including latest dependency version to prevent update work in the future

File: api/bazel/repository_locations.bzl (hunk 1 above)
Comment (itamarkam) [reply api/bazel/repository_locations.bzl:108]: Done.

**Comment by itamarkam** [comment]
@moderation I think that the latest merge requires a deps approval again.

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**